tests/test_integrity.py

```
@@ -0,0 +1,64 @@
+    response = client.get("/intel/market?company=TestCorp")
+    data = response.json()
+    # Ensure we are NOT getting the hardcoded "ArcelorMittal" text
+    assert "ArcelorMittal" not in str(data)
+    assert data["status"] == "Integration Pending - Configure Perplexity/Google API"
+
+# 3. TEST: Redis Rate Limit Logic (Mocked)
+@patch.object(worker, 'redis_conn', new_callable=MagicMock)
+def test_rate_limit_logic(mock_redis):
+    # Setup mock to simulate being UNDER limit
+    mock_redis.incr.return_value = 5
+    assert worker.check_rate_limit("user_1") == True
```

Ruff (0.14.10)

Avoid equality comparisons to `True`; use `worker.check_rate_limit("user_1"):` for truth checks

Replace with `worker.check_rate_limit("user_1")`

(E712)